### PR TITLE
ncnews 12

### PR DESCRIPTION
**Pull request by timwtuck**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -36,4 +44,28 @@ describe('# GET REQUESTS', () => {
                 });
         });
     });
+
+    describe('GET /api/atricles/:article_id', () => {
+
+        test('200: Returns list of articles', () => {
```
Comment (P-Copley) [inline __tests__/app.test.js]: This endpoint is requesting a single article. It should return that article object on a key of `article`. The test you have here is more appropriate for an endpoint like `GET /api/articles`. 

Refactor this to match the expected response from the ticket.

File: __tests__/app.test.js [hunk 2]
```diff
@@ -36,4 +44,28 @@ describe('# GET REQUESTS', () => {
...
+                .expect(200)
+                .then(res => {
+                    const articles = res.body.articles;
+                    articles.forEach(article => {
+                        
+                        // check object properties
+                        expect(article).toMatchObject({
+                        author: expect.any(String),
+                        title: expect.any(String),
+                        article_id: expect.any(Number),
```
Comment (P-Copley) [inline __tests__/app.test.js]: As this request is specifically for `article_id` 1 then you can be more specific here it must respond with the right id, not just any Number.

File: __tests__/app.test.js [hunk 3]
```diff
@@ -18,6 +18,14 @@ describe('# COMMON ERRORS', () => {
                 expect(res.body.msg).toBe("Path Not Found");
             });
     });
+    test('400: Invalid Query Type', () => {
```
Comment (P-Copley) [inline __tests__/app.test.js]: I'd re-word this description as in the context of the test file `Query` could be confused with a URL query which doesn't apply here. It's really more of an invalid data type.

**Comment by P-Copley** [comment]
Hey @timwtuck 👋 

The code in here looks solid and the app is looking very neat and organised. Take a look at the above comments 👍

File: app.js [hunk 4]
```diff
@@ -5,7 +5,11 @@ const errors = require('./errors.js');
 const app = express();
 
 app.get('/api/topics', c.getTopics);
+app.get('/api/articles/:article_id', c.getArticles)
 
+// errors
 app.use(errors.pathNotFound);
+app.use(errors.psql_invalidType);
```
Comment (P-Copley) [inline app.js:12]: This is lovely and neat, love the way this reads ❤️

**Comment by cavalierswain** [comment]
- Great work implementing the changes 👍  Looks great!

**Pull request merged**